### PR TITLE
Clarify and simplify the integration tests setup

### PR DESCRIPTION
**Pull request by geoand**
The integration tests themselves don't depend on the code of the modules
being tested.
They do however depend on those modules being deployed to Openshift.
In order for FMP to be able to deploy the modules, it needs to be able
to create an uber-resource Openshift file that contains the Openshift
definitions of both modules. That is done using the "aggregate" profile,
but assumes that FMP can find the resource file for each module on it's
classpath.
That is why the configuration has been changed to make the modules
dependencies of FMP instead of the tests themselves.

### CODE REVIEW

File: tests/pom.xml [hunk 1]
```diff
@@ -51,14 +51,10 @@
...
-      <groupId>io.openshift.booster</groupId>
-      <artifactId>spring-boot-circuit-breaker-name</artifactId>
-      <version>${project.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>io.openshift.booster</groupId>
-      <artifactId>spring-boot-circuit-breaker-greeting</artifactId>
-      <version>${project.version}</version>
+      <groupId>org.slf4j</groupId>
+      <artifactId>jcl-over-slf4j</artifactId>
```
Comment (metacosm) [inline tests/pom.xml]: Why is this needed?

File: tests/pom.xml [hunk 2]
```diff
@@ -100,38 +96,25 @@
...
+            <!--
+               We add the dependencies to the two modules that need to be deployed for the tests
+               to run.
+               This is done in order to make FMP pick up the generated Openshift
+               resources file of each module and create an "uber" resources file containing
+               resources for both modules (which is what the aggregate profile above does)
+            -->
+            <dependencies>
+              <dependency>
+                <groupId>io.openshift.booster</groupId>
```
Comment (metacosm) [inline tests/pom.xml:112]: I guess FMP (or is it Arquillian) doesn't care about the jar layout then?

File: tests/pom.xml (hunk 1 above)
Comment (geoand) [reply tests/pom.xml]: I needed to add this because I was getting NoClassDefFound exceptions without it from RestAssured...
If you have any better ideas than adding this dependency, I would love to hear them

File: tests/pom.xml (hunk 2 above)
Comment (geoand) [reply tests/pom.xml:112]: I am not certain, but my guess is that that FMP looks into the standard locations in the `target` directory where the Kubernetes / Openshift resources are placed and not the actual jar itself

File: tests/pom.xml (hunk 1 above)
Comment (gytis) [reply tests/pom.xml]: This is very weird and I don't remember needing this. But if you need to add it, then maybe add it to the dependencyManagement of the parent to not need the version in here.

Comment (geoand) [reply tests/pom.xml]: I will see if there is any way of dropping it and if not, will add it to the `dependencyManagement` section

Comment (geoand) [reply tests/pom.xml]: There is definitely something weird going on the dependency chain of rest-assured. I'm trying to get to the bottom of it

Comment (geoand) [reply tests/pom.xml]: Finally got to the bottom of this. See the changes in my second commit

**Review by metacosm — APPROVED** [review]

**Comment by geoand** [comment]
@gytis If you approve as well, I will update the other multi-module boosters as well

**Comment by metacosm** [comment]
I would like us to figure out the exact mechanism used by Arquillian / FMP to get the info it needs because the artifacts that are imported here are Spring Boot repackaged ones, so with a specific layout and I would like to document that part.

**Comment by geoand** [comment]
I'll look into it and see what I can find. I am pretty sure however that Spring Boot repackaging does not affect us here. 
I'll perform a few tests to see what I can come up with

**Review by gytis — APPROVED** [review]
Looks good, just I would like to have commons-logging defined in the parent's dependencyManagement

**Comment by geoand** [comment]
By debugging an FMP build, I was able to figure our that the component that handles figuring out what resources to use, is here: https://github.com/fabric8io/fabric8-maven-plugin/blob/v3.5.40/enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/DependencyEnricher.java

The part that interests us in this discussion is that when the Enricher is invoked for the `tests` module, looks inside the jars that are either dependencies of the project (which is why the tests work without this PR) itself or dependencies of the plugin (which is what this PR does). 
And it looks for standard files containing Kubernetes / Openshift resources:

```
    private static String DEPENDENCY_KUBERNETES_YAML = "META-INF/fabric8/kubernetes.yml";
    private static String DEPENDENCY_KUBERNETES_TEMPLATE_YAML = "META-INF/fabric8/k8s-template.yml";
    private static String DEPENDENCY_OPENSHIFT_YAML = "META-INF/fabric8/openshift.yml";
```

Spring Boot repackaging does not affect us in this case, because the repackaged jar contains the `META-INF` directory.
The jar that is created from the `tests` module (and which contains all the located resource files) is then used by Arquillian in order to deploy the application(s) to the cluster before executing the test code

**Comment by geoand** [comment]
@gytis sure thing

**Comment by metacosm** [comment]
@geoand thank you for figuring things out, this will definitely need to go in our testing guide! ;)

**Comment by geoand** [comment]
You are welcome @metacosm!

I plan to add all that info to the guide (which I need to start writing 😋)

**Comment by geoand** [comment]
The guide is being developed [here](https://raw.githubusercontent.com/snowdrop/snowdrop-docs/it/site/guides/integration-testing.adoc)

**Comment by geoand** [comment]
And the relevant PR is now [here](https://github.com/snowdrop/snowdrop-docs/pull/2)
